### PR TITLE
gui conf file: encode unicode strings as UTF-8 and read them back too

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/gui/conf/file.py [hunk 1]
```diff
@@ -119,9 +122,11 @@ def set_many(self, section, option_value_list):
         self.write()
 
     def get(self, section, option):
-        """ Get the value of an option """
+        """ Get the value of an option
+        returns (unicode): the value is converted from UTF-8
+        """
```
Comment (Sabrina85) [inline src/odemis/gui/conf/file.py]: also add the other params to the doc string

File: src/odemis/gui/conf/file.py [hunk 2]
```diff
@@ -103,10 +102,14 @@ def use_default(self):
         self.config = self.default
 
     def set(self, section, option, value):
-        """ Set the value of an option """
+        """ Set the value of an option
+        value (string or unicode): if unicode, it's encoded as UTF-8
+        """
```
Comment (Sabrina85) [inline src/odemis/gui/conf/file.py]: also add the other params to the doc string

File: src/odemis/gui/conf/test/conf_test.py [hunk 3]
```diff
@@ -138,6 +150,14 @@ def test_save(self):
...
         conf.last_format = "HDF5"
         conf.last_extension = ".h5"
+        conf.fn_ptn = u"{timelng}-test {cnt}"
+
+    def test_save_unicode(self):
+        conf = gui.conf.get_acqui_conf()
+        conf.last_path = u"/home/boooµ/"
+        conf.last_format = "HDF5"
+        conf.last_extension = ".h5"
+        conf.fn_ptn = u"{timelng}-test {cnt} µm value"
```
Comment (Sabrina85) [inline src/odemis/gui/conf/test/conf_test.py]: what is here the feedback if the test was successful?

File: src/odemis/gui/conf/file.py [hunk 4]
```diff
@@ -119,9 +122,11 @@ def set_many(self, section, option_value_list):
         self.write()
 
     def get(self, section, option):
-        """ Get the value of an option """
+        """ Get the value of an option
+        returns (unicode): the value is converted from UTF-8
+        """
         try:
-            return self.config.get(section, option)
+            return self.config.get(section, option).decode("utf-8", "replace")
```
Comment (Sabrina85) [inline src/odemis/gui/conf/file.py]: maybe add: U+FFFD � REPLACEMENT CHARACTER characters will be inserted for any bytes that can't be decoded

File: src/odemis/gui/conf/test/conf_test.py (hunk 3 above)
Comment (pieleric) [reply src/odemis/gui/conf/test/conf_test.py]: Actually, the attributes are properties, so setting them will call a function, which will fail if something goes wrong, and eventually the test case will fail.
I've added a comment explaining this.

Comment (Sabrina85) [reply src/odemis/gui/conf/test/conf_test.py]: ah ok. Does it make sense to add this `# Will fail if setting the properties goes wrong` to both testcases?

**Pull request merged**